src/app/features/group/group-list/group-list.service.ts

```diff
@@ -0,0 +1,27 @@
+import { HttpClient } from '@angular/common/http';
+import { inject, Injectable } from '@angular/core';
+import { API_URL } from '@core/env.token';
+import { BehaviorSubject, catchError, of, retry } from 'rxjs';
+import { GroupListItem } from '../group.interface';
+import { GroupListError } from '../group.interface';
+
+@Injectable()
+export class GroupListService {
+  private http = inject(HttpClient);
+  private error$$ = new BehaviorSubject<GroupListError>({ isError: false, errorStatus: 404 });
+  private apiUrl = inject(API_URL);
```

Comment:
uporządkujmy to, najpierw zależności wstrzykiwane potem lokalne pola, możemy je nawet oddzielić linijką przerwy

Reply:
Done.